tests/unit/test_bruteforce.cpp

```diff
@@ -1238,13 +1238,8 @@ TYPED_TEST(BruteForceTest, preferAdHocOptimization) {
     ASSERT_TRUE(VecSimIndex_PreferAdHocSearch(index, 0, 50, true));
 
     // Corner cases - subset size is greater than index size.
-    try {
-        VecSimIndex_PreferAdHocSearch(index, 1, 50, true);
-        FAIL() << "Expected std::runtime error";
-    } catch (std::runtime_error const &err) {
-        EXPECT_EQ(err.what(),
-                  std::string("internal error: subset size cannot be larger than index size"));
-    }
+    ASSERT_NO_THROW(VecSimIndex_PreferAdHocSearch(index, 1, 50, true));
```

Comment:
Assert that it is equals to the result with subsetSize = indexSzie rather than just assert no throw...